### DISCUSSION TITLE
Obtain earthquake source parameters for the 2020 Mw7.0 Samos earthquake

### DISCUSSION THREAD

**Discussion by fabian-kutschera** [SeismicSource/sourcespec#14] [Q&A] [answered]
Hi,

I would like to use SourceSpec to investigate earthquake source parameters like seismic moment, stress drop, etc., for the 2020 Mw7.0 Samos earthquake and compare them with reference values for the event.

I selected station HL.ARG which has a distance of ~223 km from the source and proceeded as follows:

1. Download data:
- `wget "https://service.iris.edu/fdsnws/dataselect/1/query?net=HL&sta=ARG&starttime=2020-10-30T00:00:00&endtime=2020-10-31T00:00:00&format=miniseed&nodata=404" -O "tracesHL_ARG.mseed"`
- `wget "https://service.iris.edu/fdsnws/station/1/query?net=HL&sta=ARG&starttime=2020-10-30T00:00:00&endtime=2020-10-31T00:00:00&level=response&format=xml&lat=37.897&lon=26.784&maxradius=2.000&includecomments=true&nodata=404" -O "stationHL_ARG.xml"`
- `wget "https://service.iris.edu/fdsnws/event/1/query?starttime=2020-10-30T00:00:00&endtime=2020-10-31T00:00:00&minmag=6&orderby=time&format=xml&nodata=404" -O "event.xml"`

2. Then I followed https://sourcespec.readthedocs.io/en/stable/getting_started.html#for-the-impatient
- run `source_spec -S`
- adjust `station_metadata = "stationHL_ARG.xml"`
- run `source_spec -t tracesHL_ARG.mseed -q event.xml`

These are the results I get:
![11331986 traces](https://user-images.githubusercontent.com/82662056/191348265-fd5eb3f1-2396-418c-98be-99189fe7ea68.png)
![11331986 ssp](https://user-images.githubusercontent.com/82662056/191348227-964bc184-af95-47e4-a8ec-8317e8e830fa.png)
![11331986 boxplot](https://user-images.githubusercontent.com/82662056/191348175-409e75e4-17ce-42d0-8bcd-7b1bf87b356d.png)

The theoretical P-wave pick based on the iasp91 model seems fine for me. But there is a large discrepancy in the computed moment magnitude of 4.51 compared to the real Mw7.0. 

How would I need to adjust the input parameters for `source_spec` in order to get a result which fits better? Would it matter a lot, if I manually pick the P- and S-wave arrivals (and e.g., store them in a .SAC file)?

Best regards
Fabian

**Comment by claudiodsf** [comment, marked as answer]
Thank you for inaugurating this Discussion section!

The default parameters obtained through `source_spec -S` are suited for M<5 earthquakes recorded within ~100 km (~~I will specify this in the documentation~~ Edit: [it is now specified](https://sourcespec.readthedocs.io/en/latest/getting_started.html)).

Here you want to analyse a M7 event at 225 km, so you need to:
1. use longer time windows
2. go to lower frequencies

I made a quick test by taking a signal window of 100 s and a minimum frequency of 1e-4 Hz.
The parameters to change are:
- `win_length = 100.0`
- `noise_pre_time = 120.0` (important to make sure that the noise window is well before the P arrival)
- `bp_freqmin_broadb = 1e-4`
- `freq1_broadb = 1e-2`

Edit: Note that the spectral window will start at 1/win_length = 1e-2 Hz (minimum frequency), even if `freq1_broadb` is smaller than that value.

Here are the results.

Note that the trace in the plot is filtered according to the frequencies provided in the config file and therefore it looks different from your trace (~~I will add in a future version the filter bands on the plot, to make this clear~~ Edit: [added in development version](https://github.com/SeismicSource/sourcespec/commit/d27cc1b7533c2c243c36042941ddc5db3264bb64).). The theoretical P and S picks are accurate enough, compared to the window length.

![11331986 traces](https://user-images.githubusercontent.com/1390993/191432247-d43aa2b8-6c9c-44dd-aaf8-a219e3ce7f31.png)
![11331986 ssp](https://user-images.githubusercontent.com/1390993/191432310-cb602b4d-3bcc-45cb-8578-3de5d1e601b8.png)

The estimated magnitude is Mw 6.9. Maybe you can go a little more lower in frequencies and see what happens.

As a bonus, here's a map, generated with the `plot_station_map = True` option.

![11331986 map_mag](https://user-images.githubusercontent.com/1390993/191432659-b9602a87-ec72-43c7-82d9-16b0c8de14b5.png)

Two more advices:
- Try to use more than one station
- Using shorter traces (yours is 1-hour long) will make the code faster

**Comment by fabian-kutschera** [comment]
Thank you for your reply and apologies for my late one, I first wanted to understand the paramters better and try a few things.

Your answer has been really helpful! I adjusted the parameters as you recommended but set ` win_length=80` and obtained a moment magnitude of exactly 7.00 (see figures attached). 

![11331986 map_mag](https://user-images.githubusercontent.com/82662056/191521993-bf6c76c7-ebc3-484b-b73c-4ea927b2e378.png)
![11331986 ssp](https://user-images.githubusercontent.com/82662056/191522004-646908a3-31e8-4400-9477-70f689a4d164.png)

One additional comment. I requested the event.xml file using the IRIS DMC Web Services, but I think their hypocentre depth is too deep (21 km vs 8.2 km from Cetin et al. 2020). I updated my event.xml file accordingly and found that the result is depth-insensitive, i.e., I still obtain the same estimates of the source parameters (this might of course be different for seismograms, which are much closer to the event I assume). 
Next I plan to add more stations, thanks again!

**Reply by claudiodsf** [reply]
Glad you're getting familiar with the parameters.
And, please, do not apologise for late reply: this is not a real-time help desk! 😛 

The fact that you get exactly Mw 7.0 with 80 s of window is probably a coincidence, since the code picks a lower corner frequency (respect to my result) to accommodate for the short low-frequency plateau.

My advice is to go as low-frequency as possible, so that the spectral plateau is clear and the code can better fit a Brune's model.

Concerning event depth, you are right: it has a marginal effect on hypocentral distance, given the large station distance.

**Reply by zhangzs1989** [reply]
Is the Brune model applicable to large earthquakes, such as the Mw7.0 Samos earthquake

**Reply by claudiodsf** [reply]
The Brune model can be applicable, in principle.

The problem is whether regional or teleseismic spectra can capture the very low-frequency plateau carrying information on the seismic moment.

See [Lay et al. (Tectonophysics, 2017)](https://github.com/SeismicSource/sourcespec/files/9731286/Lay-Tectonophysics-2017.pdf) for an interesting discussion on the matter (section 5, figure 9).
